internal/controller/vrg_volsync.go

```diff
@@ -639,3 +636,47 @@ func (v *VRGInstance) doCleanupResources(name, namespace string) error {
 
 	return nil
 }
+
+func (v *VRGInstance) noProtectedVolSyncPVCs() bool {
+	return len(v.volSyncPVCs) == 0 && len(v.instance.Spec.VolSync.RDSpec) == 0
+}
+
+func (v *VRGInstance) handleNoProtectedVolSyncPVCs() *metav1.Condition {
+	return newVRGAsDataProtectedUnusedCondition(v.instance.Generation,
+		"No PVCs are protected using Volsync scheme")
+}
+
+func (v *VRGInstance) handleSecondaryVolsyncPVCConflict() *metav1.Condition {
+	if v.validateSecondaryPVCConflictForVolsync() {
+		return newVRGAsDataProtectedConditionPVCConflict(v.instance.Generation,
+			"A PVC that is not a replication destination should not match the label selector.")
+	}
+
+	return newVRGAsDataProtectedUnusedCondition(v.instance.Generation,
+		"Volsync based PVC protection does not report DataProtected/ClusterDataProtected conditions as Secondary")
+}
+
+func (v *VRGInstance) validateSecondaryPVCConflictForVolsync() bool {
+	if len(v.volSyncPVCs) == 0 {
+		return false
+	}
+
+	// Validate that only replication destination PVCs match the label selector.
+	for _, pvc := range v.volSyncPVCs {
+		matchFound := false
+
+		for _, rdSpec := range v.instance.Spec.VolSync.RDSpec {
```

Comment:
Need @BenamarMk to check if this is going to an correct check all the time. There is a certain duration when the vrg is secondary and doesn't have the rdspec filled it. It might lead to spurious errors.